Packages/vcs/Lib/Canvas.py

```diff
@@ -4812,6 +4812,9 @@ def png(self, file, width=None, height=None,
     a.plot(array)
     a.png('example')       # Overwrite a png file
 """
+        base = os.path.dirname(file)
```

Comment:
Why not put this block of code in try catch instead of checking of the file exists? In general thats a better practice